pkg/resourceprovider/provider.go

```diff
@@ -159,7 +205,12 @@ func (p *resourceProvider) GetPodMetrics(pods ...*metav1.PartialObjectMetadata)
 	resultsByNs := make(map[string][]nsQueryResults, len(podsByNsBatched))
 	for result := range resChan {
 		if result.err != nil {
+			// Log the error, increment the counter, and continue
 			klog.Errorf("unable to fetch metrics for pods in namespace %q, skipping: %v", result.namespace, result.err)
+
+			// Increment the counter with namespace and error as labels
+			queryFailureCounter.WithLabelValues(result.namespace, result.err.Error()).Inc()
```

Comment:
What additional info is available in the query response? Ideally, we'd also have the error code in both metrics/logs, and also log the entire query.